openhtf/__init__.py

```diff
@@ -239,7 +239,10 @@ def execute(self, test_start=None):
             _LOG.exception(
                 'Output callback %s raised; continuing anyway', output_cb)
       finally:
+        result = (self._executor.test_state.test_record.outcome
+                  == test_record.Outcome.PASS)
         self._executor = None
+        return result
```

Comment:
returning from inside a finally block is dangerous.  specifically, you swallow any raised exceptions mysteriously and return a value instead.  theoretically this shouldn't raise anyway, but it's pretty risky.

instead, I would de-dent to be outside all the finally blocks, and
```py
return final_state.test_record.outcome == test_record.Outcome.PASS
```